src/clr-angular/utils/popover/adapter/operators.ts

```diff
@@ -0,0 +1,123 @@
+ */
+import { ClrPopoverPosition } from '../interfaces/popover-position.interface';
+import { ClrAxis } from '../enums/axis.enum';
+import { ClrSide } from '../enums/side.enum';
+import { ClrAlignment } from '../enums/alignment.enum';
+
+export const SIMPLE_TO_SMART_POSITIONS: { [input: string]: ClrPopoverPosition } = {
+  'top-left': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'top-middle': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'top-right': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'right-top': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'right-middle': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'right-bottom': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'bottom-right': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'bottom-middle': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'bottom-left': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'left-bottom': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'left-middle': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'left-top': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  default: {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+};
+
+export function transposePositionString(position: ClrPopoverPosition): string {
+  for (const key in SIMPLE_TO_SMART_POSITIONS) {
+    if (positionsEqual(position, SIMPLE_TO_SMART_POSITIONS[key])) {
+      return key;
+    }
+  }
+  return null;
+}
+
+export function transposeStringPosition(position: string): ClrPopoverPosition {
+  return SIMPLE_TO_SMART_POSITIONS[position];
+}
+
+export function validPosition(position: string): boolean {
+  return position && !!SIMPLE_TO_SMART_POSITIONS[position];
+}
+
+export function positionsEqual(pos1: ClrPopoverPosition, pos2: ClrPopoverPosition): boolean {
+  if (pos1.anchor !== pos2.anchor) {
```

Comment:
As Bozhidar suggested, this could go in one if. Looking at it here again, I agree.

Reply:
It could just keep in mind my unit-test :)